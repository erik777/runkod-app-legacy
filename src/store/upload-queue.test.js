import reducer, {
  setAct,
  startAct,
  okAct,
  errorAct,
  finishAct,
  resetAct
} from "./upload-queue"


describe('Scenario 1 - Ends up with one fail', () => {
  let state = undefined;

  it('1- default state', () => {
    expect(reducer(state, {})).toMatchSnapshot();
  });

  it('2- setAct - add files to queue. 4 items in "pending".', () => {
    const files = [
      {_id: 'id1', path: '', name: 'style.css'},
      {_id: 'id2', path: '', name: 'main.js'},
      {_id: 'id4', path: '', name: 'index.html'},
      {_id: 'id7', path: 'images/', name: 'bg.jpg'},
    ];
    const act = setAct(files);
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('3- startAct', () => {
    const act = startAct();
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('4- okAct - 3 items in "pending", 1 item in "completed".', () => {
    const act = okAct({_id: 'id2', path: '', name: 'main.js'});
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('5- okAct - 2 item in "pending", 2 items in "completed".', () => {
    const act = okAct({_id: 'id7', path: 'images/', name: 'bg.jpg'});
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('6- errorAct. 1 item in "pending", 2 items in "completed", 1 item in "failed". ', () => {
    const act = errorAct({_id: 'id4', path: '', name: 'index.html'});
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('7- okAct. 0 item in "pending", 3 items in "completed", 1 item in "failed". ', () => {
    const act = okAct({_id: 'id1', path: '', name: 'style.css'});
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('8- finishAct', () => {
    const act = finishAct();
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('9- resetAct', () => {
    const act = resetAct();
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

});


describe('Scenario 2 - All success', () => {
  let state = undefined;

  it('1- default state', () => {
    expect(reducer(state, {})).toMatchSnapshot();
  });

  it('2- setAct - add files to queue. 4 items in "pending".', () => {
    const files = [
      {_id: 'id1', path: '', name: 'style.css'},
      {_id: 'id2', path: '', name: 'main.js'},
      {_id: 'id4', path: '', name: 'index.html'},
      {_id: 'id7', path: 'images/', name: 'bg.jpg'},
    ];
    const act = setAct(files);
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('3- startAct', () => {
    const act = startAct();
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('4- okAct - 3 items in "pending", 1 item in "completed".', () => {
    const act = okAct({_id: 'id2', path: '', name: 'main.js'});
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('5- okAct - 2 item in "pending", 2 items in "completed".', () => {
    const act = okAct({_id: 'id7', path: 'images/', name: 'bg.jpg'});
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('6- okAct. 1 item in "pending", 3 items in "completed"', () => {
    const act = okAct({_id: 'id4', path: '', name: 'index.html'});
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('7- okAct. 0 item in "pending", 4 items in "completed"', () => {
    const act = okAct({_id: 'id1', path: '', name: 'style.css'});
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('8- finishAct', () => {
    const act = finishAct();
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

  it('9- resetAct', () => {
    const act = resetAct();
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
  });

});

