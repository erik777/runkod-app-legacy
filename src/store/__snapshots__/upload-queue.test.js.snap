// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scenario 1 - Ends up with one fail 1- default state 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "inProgress": false,
  "log": Array [],
  "pending": Array [],
  "show": false,
}
`;

exports[`Scenario 1 - Ends up with one fail 2- setAct - add files to queue. 4 items in "pending". 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "inProgress": false,
  "log": Array [],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 1 - Ends up with one fail 3- startAct 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
  ],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 1 - Ends up with one fail 4- okAct - 3 items in "pending", 1 item in "completed". 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
  ],
  "failed": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
  ],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 1 - Ends up with one fail 5- okAct - 2 item in "pending", 2 items in "completed". 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "failed": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
    Object {
      "msg": "/images/bg.jpg uploaded",
      "type": "success",
    },
  ],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 1 - Ends up with one fail 6- errorAct. 1 item in "pending", 2 items in "completed", 1 item in "failed".  1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "failed": Array [
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
  ],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
    Object {
      "msg": "/images/bg.jpg uploaded",
      "type": "success",
    },
    Object {
      "msg": "/index.html could not upload",
      "type": "error",
    },
  ],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 1 - Ends up with one fail 7- okAct. 0 item in "pending", 3 items in "completed", 1 item in "failed".  1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
  ],
  "failed": Array [
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
  ],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
    Object {
      "msg": "/images/bg.jpg uploaded",
      "type": "success",
    },
    Object {
      "msg": "/index.html could not upload",
      "type": "error",
    },
    Object {
      "msg": "/style.css uploaded",
      "type": "success",
    },
  ],
  "pending": Array [],
  "show": true,
}
`;

exports[`Scenario 1 - Ends up with one fail 8- finishAct 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
  ],
  "failed": Array [
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
  ],
  "inProgress": false,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
    Object {
      "msg": "/images/bg.jpg uploaded",
      "type": "success",
    },
    Object {
      "msg": "/index.html could not upload",
      "type": "error",
    },
    Object {
      "msg": "/style.css uploaded",
      "type": "success",
    },
    Object {
      "msg": "3 files uploaded, 1 failed  ðŸ‘€",
      "type": "info",
    },
  ],
  "pending": Array [],
  "show": true,
}
`;

exports[`Scenario 1 - Ends up with one fail 9- resetAct 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "inProgress": false,
  "log": Array [],
  "pending": Array [],
  "show": false,
}
`;

exports[`Scenario 2 - All success 1- default state 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "inProgress": false,
  "log": Array [],
  "pending": Array [],
  "show": false,
}
`;

exports[`Scenario 2 - All success 2- setAct - add files to queue. 4 items in "pending". 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "inProgress": false,
  "log": Array [],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 2 - All success 3- startAct 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
  ],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 2 - All success 4- okAct - 3 items in "pending", 1 item in "completed". 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
  ],
  "failed": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
  ],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 2 - All success 5- okAct - 2 item in "pending", 2 items in "completed". 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
  ],
  "failed": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
    Object {
      "msg": "/images/bg.jpg uploaded",
      "type": "success",
    },
  ],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 2 - All success 6- okAct. 1 item in "pending", 3 items in "completed" 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
  ],
  "failed": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
    Object {
      "msg": "/images/bg.jpg uploaded",
      "type": "success",
    },
    Object {
      "msg": "/index.html uploaded",
      "type": "success",
    },
  ],
  "pending": Array [
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
  ],
  "show": true,
}
`;

exports[`Scenario 2 - All success 7- okAct. 0 item in "pending", 4 items in "completed" 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
  ],
  "failed": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
    Object {
      "msg": "/images/bg.jpg uploaded",
      "type": "success",
    },
    Object {
      "msg": "/index.html uploaded",
      "type": "success",
    },
    Object {
      "msg": "/style.css uploaded",
      "type": "success",
    },
  ],
  "pending": Array [],
  "show": true,
}
`;

exports[`Scenario 2 - All success 8- finishAct 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "id2",
      "name": "main.js",
      "path": "",
    },
    Object {
      "_id": "id7",
      "name": "bg.jpg",
      "path": "images/",
    },
    Object {
      "_id": "id4",
      "name": "index.html",
      "path": "",
    },
    Object {
      "_id": "id1",
      "name": "style.css",
      "path": "",
    },
  ],
  "failed": Array [],
  "inProgress": false,
  "log": Array [
    Object {
      "msg": "Upload started",
      "type": "info",
    },
    Object {
      "msg": "/main.js uploaded",
      "type": "success",
    },
    Object {
      "msg": "/images/bg.jpg uploaded",
      "type": "success",
    },
    Object {
      "msg": "/index.html uploaded",
      "type": "success",
    },
    Object {
      "msg": "/style.css uploaded",
      "type": "success",
    },
    Object {
      "msg": "4 files uploaded ðŸŽ‰",
      "type": "info",
    },
  ],
  "pending": Array [],
  "show": true,
}
`;

exports[`Scenario 2 - All success 9- resetAct 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "inProgress": false,
  "log": Array [],
  "pending": Array [],
  "show": false,
}
`;
